pinot-common/src/main/java/org/apache/pinot/common/config/SegmentsValidationAndRetentionConfig.java

```diff
@@ -86,12 +88,17 @@ public void setTimeColumnName(String timeColumnName) {
     this.timeColumnName = timeColumnName;
   }
 
-  public String getTimeType() {
-    return timeType;
+  public TimeUnit getTimeType() {
+    return _timeType;
+  }
+
+  @JsonIgnore
+  public void setTimeType(TimeUnit timeType) {
```

Comment:
I guess this is for completeness? (seems to be not used)

Reply:
Actually this is the cause of NPE... Removed